sycl/source/detail/platform_impl.cpp

```diff
@@ -67,6 +177,10 @@ platform_impl_pi::get_devices(info::device_type deviceType) const {
   PI_CALL(piDevicesGet)(m_platform, pi::cast<RT::PiDeviceType>(deviceType),
                         num_devices, pi_devices.data(), nullptr);
 
+  // Filter out devices that are not present in the white list
+  if (SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get())
+    filterWhiteList(pi_devices);
```

Comment:
Is SYCLConfig caches the configuration results?
If not, it makes sense to re-use it:
```c++
if (const char *WhiteList = SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get())
  filterWhiteList(pi_devices, WhiteList);
```

Reply:
It caches results.